lib/vector-core/src/fanout.rs

```diff
@@ -42,13 +37,80 @@ impl Fanout {
 
         let fanout = Self {
             sinks: vec![],
-            i: 0,
             control_channel: control_rx,
         };
 
         (fanout, control_tx)
     }
 
+    /// Send a full vec of Events into the Fanout
+    ///
+    /// This function accepts a `Vec<Event>` and duplicates it down into the
+    /// Fanout's sinks. We take care to avoid the slow receiver problem as much
+    /// as possible but any misbehaving downstream WILL cause this function to
+    /// suffer high latencies. The memory use of this function is
+    /// `N*mem::size_of(events)`, where N is the number of downstream targets.
+    ///
+    /// # Errors
+    ///
+    /// Function will error with `()` in case downstream Sink operations
+    /// fail. This is, we recognize, not very helpful.
+    ///
+    /// # Panics
+    ///
+    /// This function will only panic if there are internal logic errors,
+    /// specifically around the size of the buffer created to house clones of
+    /// Vec<Event>.
+    pub async fn send_all(&mut self, events: Vec<Event>) -> Result<(), ()> {
+        self.process_control_messages().await;
+
+        if self.sinks.is_empty() {
+            // nothing, intentionally
+        } else if self.sinks.len() == 1 {
+            if let Some(sink) = &mut self.sinks[0].1 {
+                for event in events {
+                    sink.feed(event).await?;
+                }
+                sink.flush().await?;
+            }
+        } else {
+            let mut faulty_sinks = Vec::with_capacity(0);
```

Comment:
Um, is `Vec::with_capacity(0)` not just `Vec::new`?

Reply:
It is. I'll adjust this.